src/components/profile/friend-list.component.tsx

```diff
@@ -153,9 +161,10 @@ const FriendsList = ({ profile }: Props) => {
     setOpenModal(!openModal);
   };
 
-  const handleUnFriendRequest = (friendStatus: ExtendedFriend) => {
-    cancelFriendRequest(friendStatus);
+  const handleUnFriendRequest = (friend: ExtendedFriend | null) => {
+    if (friend) cancelFriendRequest(friend);
```

Comment:
do you mean `cancelUnFriendRequest`? kalau emang friend then then unfriend request should go through

Reply:
there is 2 function 'handleUnfriendRequest' & 'handlecancelFriendRequest' yang gunain cancelFriendRequest function.
cancelFriendRequest gunanya untuk delete friendstatus yang sudah ada.
bedanya handleUnfriendRequest: friend statusnya approved kalau, handlecancelFriendRequest: statusnya masih pending.
karena fungsinya sama, jadi cancelFriendRequest dipake dikeduanya. atau mungkin baiknya dipisah?